In [1]:
import pandas as pd
import numpy as np
import altair as alt

In [2]:
def to_list_if_array(data):
    if isinstance(data, (list, tuple, np.ndarray)):
        return list(data)
    else:
        return data

# Part 0: Data Wrangling

In [3]:
mlb = pd.read_csv("C:/Users\ericg\Downloads\MLB.csv")

In [4]:
mlb

Team Name Edit Survey link           Timestamp  \
0          Houston Astros      Edit Link 1  7/26/2023 17:33:15   
1      Kansas City Royals      Edit Link 2    6/5/2025 5:50:32   
2           New York Mets      Edit Link 3    6/5/2025 6:08:11   
3   Philadelphia Phillies      Edit Link 4    6/5/2025 6:14:42   
4      Pittsburgh Pirates      Edit Link 5    6/5/2025 8:18:27   
..                    ...              ...                 ...   
67                    NaN              NaN                 NaN   
68                    NaN              NaN                 NaN   
69                    NaN              NaN                 NaN   
70         Houston Astros      Edit Link 1    6/5/2025 5:45:15   
71       San Diego Padres     Edit Link 14  2/18/2025 14:12:36   

       Contacts Name    Phone Number Grocery Partner  \
0     Jimmy Comerota  (713) 447-0782           H-E-B   
1        David Adams    760-424-6067   Price Chopper   
2      Meghan Baker     718-559-3047           Getir   
3      Scott Nickie     215-218-5286           Giant   
4      Chris Stevens    412-260-5559     Giant Eagle   
..               ...             ...             ...   
67               NaN             NaN             NaN   
68               NaN             NaN             NaN   
69               NaN             NaN             NaN   
70    Jimmy Comerota  (713)-447-0782           H-E-B   
71  Sergio Del Prado  (619) 208-7038             NaN   

   Current Concession Company  \
0                     Aramark   
1                     Aramark   
2                     Aramark   
3                     Aramark   
4                     Aramark   
..                        ...   
67                        NaN   
68                        NaN   
69                        NaN   
70                    Aramark   
71             Delaware North   

                                        Managers Name  \
0   He felt uncomfortable during the phone call co...   
1                                       Matt Quatraro   
2                                      Carlos Mendoza   
3                                         Rob Thomson   
4   John Bray, Manager, Corporate & Premium Partne...   
..                                                ...   
67                                                NaN   
68                                                NaN   
69                                                NaN   
70                                             unsure   
71                                             unsure   

                          Food Distributor Convenience Store partner  ...  \
0                                    Sysco                       NaN  ...   
1                         Kauffman Stadium                  QuikTrip  ...   
2                               Citi Field                       NaN  ...   
3                   Giant Food Stores, LLC                     Giant  ...   
4   Greater Pittsburgh Community Food Bank     Multiple-No Exclusive  ...   
..                                     ...                       ...  ...   
67                                     NaN                       NaN  ...   
68                                     NaN                       NaN  ...   
69                                     NaN                       NaN  ...   
70                                   Sysco                       NaN  ...   
71                                   Sysco                       NaN  ...   

   Notes  question above                          Notes to question above.1  \
0                    NaN                                                NaN   
1       Done by Research                                   Done by research   
2       Done by research  Located on Field Level behind section 101 and 128   
3       Done by Research                                   Done by research   
4       Done by research                                   Done by research   
..                   ...                                             

In [5]:
mlb2 = mlb.drop(columns={'Edit Survey link'}) # Drop the survey links since they do not work

In [6]:
mlb

Team Name Edit Survey link           Timestamp  \
0          Houston Astros      Edit Link 1  7/26/2023 17:33:15   
1      Kansas City Royals      Edit Link 2    6/5/2025 5:50:32   
2           New York Mets      Edit Link 3    6/5/2025 6:08:11   
3   Philadelphia Phillies      Edit Link 4    6/5/2025 6:14:42   
4      Pittsburgh Pirates      Edit Link 5    6/5/2025 8:18:27   
..                    ...              ...                 ...   
67                    NaN              NaN                 NaN   
68                    NaN              NaN                 NaN   
69                    NaN              NaN                 NaN   
70         Houston Astros      Edit Link 1    6/5/2025 5:45:15   
71       San Diego Padres     Edit Link 14  2/18/2025 14:12:36   

       Contacts Name    Phone Number Grocery Partner  \
0     Jimmy Comerota  (713) 447-0782           H-E-B   
1        David Adams    760-424-6067   Price Chopper   
2      Meghan Baker     718-559-3047           Getir   
3      Scott Nickie     215-218-5286           Giant   
4      Chris Stevens    412-260-5559     Giant Eagle   
..               ...             ...             ...   
67               NaN             NaN             NaN   
68               NaN             NaN             NaN   
69               NaN             NaN             NaN   
70    Jimmy Comerota  (713)-447-0782           H-E-B   
71  Sergio Del Prado  (619) 208-7038             NaN   

   Current Concession Company  \
0                     Aramark   
1                     Aramark   
2                     Aramark   
3                     Aramark   
4                     Aramark   
..                        ...   
67                        NaN   
68                        NaN   
69                        NaN   
70                    Aramark   
71             Delaware North   

                                        Managers Name  \
0   He felt uncomfortable during the phone call co...   
1                                       Matt Quatraro   
2                                      Carlos Mendoza   
3                                         Rob Thomson   
4   John Bray, Manager, Corporate & Premium Partne...   
..                                                ...   
67                                                NaN   
68                                                NaN   
69                                                NaN   
70                                             unsure   
71                                             unsure   

                          Food Distributor Convenience Store partner  ...  \
0                                    Sysco                       NaN  ...   
1                         Kauffman Stadium                  QuikTrip  ...   
2                               Citi Field                       NaN  ...   
3                   Giant Food Stores, LLC                     Giant  ...   
4   Greater Pittsburgh Community Food Bank     Multiple-No Exclusive  ...   
..                                     ...                       ...  ...   
67                                     NaN                       NaN  ...   
68                                     NaN                       NaN  ...   
69                                     NaN                       NaN  ...   
70                                   Sysco                       NaN  ...   
71                                   Sysco                       NaN  ...   

   Notes  question above                          Notes to question above.1  \
0                    NaN                                                NaN   
1       Done by Research                                   Done by research   
2       Done by research  Located on Field Level behind section 101 and 128   
3       Done by Research                                   Done by research   
4       Done by research                                   Done by research   
..                   ...                                             

In [7]:
pd.options.display.max_columns=None

In [8]:
mlb2 = mlb2.rename(columns = {'Team Name':"team_name",'Timestamp':'timestamp','Contacts Name':'contact_name','Phone Number':'phone_number',
                      'Grocery Partner':'grocery_partner','Current Concession Company':'concession_corp','Managers Name':'manager_name',
                      'Food Distributor':'food_distributor','Convenience Store partner':'convenience_partner',
                      'Do you currently have sponsors on your cupholders?':'has_cuphold_sponsors','If yes, by whom?':'cuphold_sponsors',
                     'Please kindly check the items you sponsor at concession areas.':'concession_area_items','Do you have in-Stadium WIFI?':'has_wifi',
                     'If yes, who is your current partner?':'wifi_partner','Do you offer charging stations?':'has_charge_stations',
                      'If yes, sponsored by whom?':'charging_sponsors','Do you offer any of these digital signage outside the stadium':'outside_digital_signage',
                     'Do you currently offer advertisers into the stadium Tv network':'offers_ads','What are couple of your main targets for fan engagements?':'main_targets',
                     'How can veteran programs work with you and bring their message of victory':'veteran_programs',
                     'Would you consider using our free format at www.buysportsmedia.com to list excess inventory your sponsorships and other opportunities within the stadium or outside the stadium':'use_free_format',
                     'What food categories are available for sponsorship in your stadium':'available_food_categories','Who are your current pouring partners':'pouring_partners',
                     "What are some of your current non-exclusive categories that you're searching for new business":'non_exclusive_categories',
                     'What exclusives or partnerships are ending this year where there might be an opportunity to bring in a new partner':'yearly_exclusives_and_partnerships',
                     'Do you control your radio and television network':'control_radio_and_tv','If clicked on other, what is the food category':'other_food_categories',
                     'Do you offer Tv rotational signage?':'offers_rotational_signage','Who is your Wifi Carrier?':'wifi_carrier','Notes to question above':'wifi_carrier_notes',
                     'if checked Cupholders, How many seats has a cupholder in your stadium or arena':'cupholders_per_seat',
                     'if checked Souvenir cups, How many seats has a souvenir cup in your stadium or arena':'souvenir_cups_per_seat',
                     'if checked concession trays, what is your concession tray annual volume':'concession_tray_annual_volume',
                     'Notes question above':'volume_notes','What is your current charging solution?':'current_charging_station',
                     'if yes, what is your rate?':'rate','Which is requested?':'requested','Notes to question above.1':'annual_volume_notes'})

In [9]:
mlb2 = mlb2.drop(columns={'if checked Cupholders, How many seats has a cupholder in your stadium or arena.1',
                        'if checked Souvenir cups, How many seats has a souvenir cup in your stadium or arena.1',
                        'if checked concession trays, what is your concession tray annual volume.1',
                        'Who is your Wifi Carrier?.1'})

In [10]:
mlb2

team_name           timestamp      contact_name  \
0          Houston Astros  7/26/2023 17:33:15    Jimmy Comerota   
1      Kansas City Royals    6/5/2025 5:50:32       David Adams   
2           New York Mets    6/5/2025 6:08:11     Meghan Baker    
3   Philadelphia Phillies    6/5/2025 6:14:42     Scott Nickie    
4      Pittsburgh Pirates    6/5/2025 8:18:27     Chris Stevens   
..                    ...                 ...               ...   
67                    NaN                 NaN               NaN   
68                    NaN                 NaN               NaN   
69                    NaN                 NaN               NaN   
70         Houston Astros    6/5/2025 5:45:15    Jimmy Comerota   
71       San Diego Padres  2/18/2025 14:12:36  Sergio Del Prado   

      phone_number grocery_partner concession_corp  \
0   (713) 447-0782           H-E-B         Aramark   
1     760-424-6067   Price Chopper         Aramark   
2     718-559-3047           Getir         Aramark   
3     215-218-5286           Giant         Aramark   
4     412-260-5559     Giant Eagle         Aramark   
..             ...             ...             ...   
67             NaN             NaN             NaN   
68             NaN             NaN             NaN   
69             NaN             NaN             NaN   
70  (713)-447-0782           H-E-B         Aramark   
71  (619) 208-7038             NaN  Delaware North   

                                         manager_name  \
0   He felt uncomfortable during the phone call co...   
1                                       Matt Quatraro   
2                                      Carlos Mendoza   
3                                         Rob Thomson   
4   John Bray, Manager, Corporate & Premium Partne...   
..                                                ...   
67                                                NaN   
68                                                NaN   
69                                                NaN   
70                                             unsure   
71                                             unsure   

                          food_distributor    convenience_partner  \
0                                    Sysco                    NaN   
1                         Kauffman Stadium               QuikTrip   
2                               Citi Field                    NaN   
3                   Giant Food Stores, LLC                  Giant   
4   Greater Pittsburgh Community Food Bank  Multiple-No Exclusive   
..                                     ...                    ...   
67                                     NaN                    NaN   
68                                     NaN                    NaN   
69                                     NaN                    NaN   
70                                   Sysco                    NaN   
71                                   Sysco                    NaN   

   has_cuphold_sponsors cuphold_sponsors  \
0                    No              NaN   
1                    No              NaN   
2                    No              NaN   
3                   Yes          Citizen   
4                    No              NaN   
..                  ...              ...   
67                  NaN              NaN   
68                  NaN              NaN   
69                  NaN              NaN   
70                   No              NaN   
71                   No              NaN   

                                concession_area_items has_wifi  \
0   Souvenir cups, Concession trays, printed paper...      Yes   
1   Souvenir cups, Concession trays, printed paper...      Yes   
2                                       Souvenir cups      Yes   
3        cup holders, Souvenir cups, Concession trays      Yes   
4                     Souvenir cups, Concession trays      Yes   
..                                                ...      ...   
67                                                NaN    

In [11]:
mlb

Team Name Edit Survey link           Timestamp  \
0          Houston Astros      Edit Link 1  7/26/2023 17:33:15   
1      Kansas City Royals      Edit Link 2    6/5/2025 5:50:32   
2           New York Mets      Edit Link 3    6/5/2025 6:08:11   
3   Philadelphia Phillies      Edit Link 4    6/5/2025 6:14:42   
4      Pittsburgh Pirates      Edit Link 5    6/5/2025 8:18:27   
..                    ...              ...                 ...   
67                    NaN              NaN                 NaN   
68                    NaN              NaN                 NaN   
69                    NaN              NaN                 NaN   
70         Houston Astros      Edit Link 1    6/5/2025 5:45:15   
71       San Diego Padres     Edit Link 14  2/18/2025 14:12:36   

       Contacts Name    Phone Number Grocery Partner  \
0     Jimmy Comerota  (713) 447-0782           H-E-B   
1        David Adams    760-424-6067   Price Chopper   
2      Meghan Baker     718-559-3047           Getir   
3      Scott Nickie     215-218-5286           Giant   
4      Chris Stevens    412-260-5559     Giant Eagle   
..               ...             ...             ...   
67               NaN             NaN             NaN   
68               NaN             NaN             NaN   
69               NaN             NaN             NaN   
70    Jimmy Comerota  (713)-447-0782           H-E-B   
71  Sergio Del Prado  (619) 208-7038             NaN   

   Current Concession Company  \
0                     Aramark   
1                     Aramark   
2                     Aramark   
3                     Aramark   
4                     Aramark   
..                        ...   
67                        NaN   
68                        NaN   
69                        NaN   
70                    Aramark   
71             Delaware North   

                                        Managers Name  \
0   He felt uncomfortable during the phone call co...   
1                                       Matt Quatraro   
2                                      Carlos Mendoza   
3                                         Rob Thomson   
4   John Bray, Manager, Corporate & Premium Partne...   
..                                                ...   
67                                                NaN   
68                                                NaN   
69                                                NaN   
70                                             unsure   
71                                             unsure   

                          Food Distributor Convenience Store partner  \
0                                    Sysco                       NaN   
1                         Kauffman Stadium                  QuikTrip   
2                               Citi Field                       NaN   
3                   Giant Food Stores, LLC                     Giant   
4   Greater Pittsburgh Community Food Bank     Multiple-No Exclusive   
..                                     ...                       ...   
67                                     NaN                       NaN   
68                                     NaN                       NaN   
69                                     NaN                       NaN   
70                                   Sysco                       NaN   
71                                   Sysco                       NaN   

   Do you currently have sponsors on your cupholders? If yes, by whom?  \
0                                                  No              NaN   
1                                                  No              NaN   
2                                                  No              NaN   
3                                                 Yes          Citizen   
4                                                  No              NaN   
..                                                ...              ...   
67                                                NaN              NaN   
68         

In [12]:
mlb2 = mlb2.fillna("unsure")

In [13]:
mlb2

team_name           timestamp      contact_name  \
0          Houston Astros  7/26/2023 17:33:15    Jimmy Comerota   
1      Kansas City Royals    6/5/2025 5:50:32       David Adams   
2           New York Mets    6/5/2025 6:08:11     Meghan Baker    
3   Philadelphia Phillies    6/5/2025 6:14:42     Scott Nickie    
4      Pittsburgh Pirates    6/5/2025 8:18:27     Chris Stevens   
..                    ...                 ...               ...   
67                 unsure              unsure            unsure   
68                 unsure              unsure            unsure   
69                 unsure              unsure            unsure   
70         Houston Astros    6/5/2025 5:45:15    Jimmy Comerota   
71       San Diego Padres  2/18/2025 14:12:36  Sergio Del Prado   

      phone_number grocery_partner concession_corp  \
0   (713) 447-0782           H-E-B         Aramark   
1     760-424-6067   Price Chopper         Aramark   
2     718-559-3047           Getir         Aramark   
3     215-218-5286           Giant         Aramark   
4     412-260-5559     Giant Eagle         Aramark   
..             ...             ...             ...   
67          unsure          unsure          unsure   
68          unsure          unsure          unsure   
69          unsure          unsure          unsure   
70  (713)-447-0782           H-E-B         Aramark   
71  (619) 208-7038          unsure  Delaware North   

                                         manager_name  \
0   He felt uncomfortable during the phone call co...   
1                                       Matt Quatraro   
2                                      Carlos Mendoza   
3                                         Rob Thomson   
4   John Bray, Manager, Corporate & Premium Partne...   
..                                                ...   
67                                             unsure   
68                                             unsure   
69                                             unsure   
70                                             unsure   
71                                             unsure   

                          food_distributor    convenience_partner  \
0                                    Sysco                 unsure   
1                         Kauffman Stadium               QuikTrip   
2                               Citi Field                 unsure   
3                   Giant Food Stores, LLC                  Giant   
4   Greater Pittsburgh Community Food Bank  Multiple-No Exclusive   
..                                     ...                    ...   
67                                  unsure                 unsure   
68                                  unsure                 unsure   
69                                  unsure                 unsure   
70                                   Sysco                 unsure   
71                                   Sysco                 unsure   

   has_cuphold_sponsors cuphold_sponsors  \
0                    No           unsure   
1                    No           unsure   
2                    No           unsure   
3                   Yes          Citizen   
4                    No           unsure   
..                  ...              ...   
67               unsure           unsure   
68               unsure           unsure   
69               unsure           unsure   
70                   No           unsure   
71                   No           unsure   

                                concession_area_items has_wifi  \
0   Souvenir cups, Concession trays, printed paper...      Yes   
1   Souvenir cups, Concession trays, printed paper...      Yes   
2                                       Souvenir cups      Yes   
3        cup holders, Souvenir cups, Concession trays      Yes   
4                     Souvenir cups, Concession trays      Yes   
..                                                ...      ...   
67                                             unsure   u

# Food & Beverage Partner Analysis
1. 
Which food distributors and grocery partners are most common across venueons?

2. What are the most popular food categories available for sponsorship? Are there gaps (e.g., niche categories like brisket or smoked chicken)
3. Are there opportunities in categories where sponsorship is low but concession volume is high
4. What is the average stadium seat count compared to souvenir cup distribution? Can increased souvenir cups lead to more ad impressions

In [14]:
mlb2['food_distributor'] = mlb2['food_distributor'].str.replace(' ','')
# One-hot encoding with comma as a separator
dummies = mlb2['food_distributor'].str.get_dummies(sep=',')
# Concatenate the encoded columns with the original DataFrame
mlb2 = pd.concat([mlb2, dummies], axis=1)

Note: 0s represent the venues that did NOT have a specific food distributor, whereas 1s indicate that the stadium did have a specific food distributor.

In [15]:
mlb_food_dist = dummies.iloc[:, 0:len(dummies)-1]
mlb_food_dist = pd.concat([mlb_food_dist, mlb2.team_name], axis=1)
mlb_food_dist = pd.concat([mlb_food_dist,mlb2.food_distributor],axis=1)

In [16]:
value_counts1 = []
for i in range(len(mlb_food_dist.columns)-2):
    value_counts1.append(mlb2.iloc[:,i+40].value_counts())
    print(mlb2.iloc[:,i+40].value_counts())

Aramark
0    69
1     3
Name: count, dtype: int64
AramarkSports+Entertainment
0    71
1     1
Name: count, dtype: int64
BallParkBrand
0    71
1     1
Name: count, dtype: int64
ChicagoCubs'WrigleyField
0    71
1     1
Name: count, dtype: int64
CitiField
0    71
1     1
Name: count, dtype: int64
Coca-Cola
0    71
1     1
Name: count, dtype: int64
DelawareNorth
0    67
1     5
Name: count, dtype: int64
DelawareNorthCompaniesSportservice
0    71
1     1
Name: count, dtype: int64
DelawareNorthSportservice
0    71
1     1
Name: count, dtype: int64
FoodBankforNewYorkCity
0    71
1     1
Name: count, dtype: int64
FreshDirect
0    71
1     1
Name: count, dtype: int64
GiantFoodStores
0    71
1     1
Name: count, dtype: int64
GreaterPittsburghCommunityFoodBank
0    71
1     1
Name: count, dtype: int64
KUNA
0    71
1     1
Name: count, dtype: int64
KauffmanStadium
0    71
1     1
Name: count, dtype: int64
Kuna
0    71
1     1
Name: count, dtype: int64
LLC
0    71
1     1
Name: count, dtype: int64


In [17]:
charts = []
for i in range(len(value_counts1)-1):
    value_counts = value_counts1[i].reset_index()
    value_counts.columns = ['category', 'count']
    charts.append(alt.Chart(value_counts,title='Histogram of Food Sponsors').mark_bar().encode(
        x=alt.X('category:N', title='Does This Food company sponsor this stadium?'),
        y=alt.Y('count:Q', title='Count of Records'),
    ))

In [18]:
print(value_counts1[0])
charts[0]

Aramark
0    69
1     3
Name: count, dtype: int64


alt.Chart(...)

In [19]:
print(value_counts1[1])
(charts[1])

AramarkSports+Entertainment
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [20]:
print(value_counts1[2])
charts[2]

BallParkBrand
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [21]:
print(value_counts1[3])
(charts[3])

ChicagoCubs'WrigleyField
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [22]:
print(value_counts1[4])
charts[4]

CitiField
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [23]:
print(value_counts1[5])
(charts[5])

Coca-Cola
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [24]:
print(value_counts1[6])
charts[6]

DelawareNorth
0    67
1     5
Name: count, dtype: int64


alt.Chart(...)

In [25]:
print(value_counts1[7])
(charts[7])

DelawareNorthCompaniesSportservice
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [26]:
print(value_counts1[8])
charts[8]

DelawareNorthSportservice
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [27]:
print(value_counts1[9])
(charts[9])

FoodBankforNewYorkCity
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [28]:
print(value_counts1[10])
charts[10]

FreshDirect
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [29]:
print(value_counts1[11])
(charts[11])

GiantFoodStores
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [30]:
print(value_counts1[12])
charts[12]

GreaterPittsburghCommunityFoodBank
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [31]:
print(value_counts1[13])
(charts[13])

KUNA
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [32]:
print(value_counts1[14])
charts[14]

KauffmanStadium
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [33]:
print(value_counts1[15])
(charts[15])

Kuna
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [34]:
print(value_counts1[16])
charts[16]

LLC
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [35]:
print(value_counts1[17])
(charts[17])

Legends
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [36]:
print(value_counts1[18])
charts[18]

LegendsHospitality
0    70
1     2
Name: count, dtype: int64


alt.Chart(...)

In [37]:
print(value_counts1[19])
(charts[19])

LevyRestaurants
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [38]:
print(value_counts1[20])
charts[20]

LittleCaesars
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [39]:
print(value_counts1[21])
(charts[21])

Marrow
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [40]:
print(value_counts1[22])
charts[22]

PapaCantella's
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [41]:
print(value_counts1[23])
(charts[23])

SlowsBarBQ
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [42]:
print(value_counts1[24])
charts[24]

Sysco
0    67
1     5
Name: count, dtype: int64


alt.Chart(...)

In [43]:
print(value_counts1[25])
(charts[25])

andFeedingTampaBay
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [44]:
mlb2['grocery_partner'] = mlb2['grocery_partner'].str.replace(' ','')
# One-hot encoding with comma as a separator
dummies = mlb2['grocery_partner'].str.get_dummies(sep=',')
# Concatenate the encoded columns with the original DataFrame
mlb2 = pd.concat([mlb2, dummies], axis=1)

In [45]:
def drop_duplicate_columns(df):
  """Drops duplicate columns from a Pandas DataFrame.

  Args:
      df: The input DataFrame.

  Returns:
      A new DataFrame with duplicate columns removed.
  """
  df_transposed = df.T
  df_unique_columns = df_transposed.drop_duplicates()
  return df_unique_columns.T

In [46]:
mlb2 = drop_duplicate_columns(mlb2)
mlb2=mlb2.drop('unsure',axis=1)

In [47]:
mlb2

team_name           timestamp      contact_name  \
0          Houston Astros  7/26/2023 17:33:15    Jimmy Comerota   
1      Kansas City Royals    6/5/2025 5:50:32       David Adams   
2           New York Mets    6/5/2025 6:08:11     Meghan Baker    
3   Philadelphia Phillies    6/5/2025 6:14:42     Scott Nickie    
4      Pittsburgh Pirates    6/5/2025 8:18:27     Chris Stevens   
..                    ...                 ...               ...   
67                 unsure              unsure            unsure   
68                 unsure              unsure            unsure   
69                 unsure              unsure            unsure   
70         Houston Astros    6/5/2025 5:45:15    Jimmy Comerota   
71       San Diego Padres  2/18/2025 14:12:36  Sergio Del Prado   

      phone_number grocery_partner concession_corp  \
0   (713) 447-0782           H-E-B         Aramark   
1     760-424-6067    PriceChopper         Aramark   
2     718-559-3047           Getir         Aramark   
3     215-218-5286           Giant         Aramark   
4     412-260-5559      GiantEagle         Aramark   
..             ...             ...             ...   
67          unsure          unsure          unsure   
68          unsure          unsure          unsure   
69          unsure          unsure          unsure   
70  (713)-447-0782           H-E-B         Aramark   
71  (619) 208-7038          unsure  Delaware North   

                                         manager_name  \
0   He felt uncomfortable during the phone call co...   
1                                       Matt Quatraro   
2                                      Carlos Mendoza   
3                                         Rob Thomson   
4   John Bray, Manager, Corporate & Premium Partne...   
..                                                ...   
67                                             unsure   
68                                             unsure   
69                                             unsure   
70                                             unsure   
71                                             unsure   

                      food_distributor    convenience_partner  \
0                                Sysco                 unsure   
1                      KauffmanStadium               QuikTrip   
2                            CitiField                 unsure   
3                  GiantFoodStores,LLC                  Giant   
4   GreaterPittsburghCommunityFoodBank  Multiple-No Exclusive   
..                                 ...                    ...   
67                              unsure                 unsure   
68                              unsure                 unsure   
69                              unsure                 unsure   
70                               Sysco                 unsure   
71                               Sysco                 unsure   

   has_cuphold_sponsors cuphold_sponsors  \
0                    No           unsure   
1                    No           unsure   
2                    No           unsure   
3                   Yes          Citizen   
4                    No           unsure   
..                  ...              ...   
67               unsure           unsure   
68               unsure           unsure   
69               unsure           unsure   
70                   No           unsure   
71                   No           unsure   

                                concession_area_items has_wifi  \
0   Souvenir cups, Concession trays, printed paper...      Yes   
1   Souvenir cups, Concession trays, printed paper...      Yes   
2                                       Souvenir cups      Yes   
3        cup holders, Souvenir cups, Concession trays      Yes   
4                     Souvenir cups, Concession trays      Yes   
..                                                ...      ...   
67                                             unsure   unsure   
68                                     

In [48]:
dummies

AramarkSports+Entertainment  BonAppétitManagementCompany  CHSInc  \
0                             0                            0       0   
1                             0                            0       0   
2                             0                            0       0   
3                             0                            0       0   
4                             0                            0       0   
..                          ...                          ...     ...   
67                            0                            0       0   
68                            0                            0       0   
69                            0                            0       0   
70                            0                            0       0   
71                            0                            0       0   

    Chick-Fli-A  Coca-Cola  DelawareNorth  \
0             0          0              0   
1             0          0              0   
2             0          0              0   
3             0          0              0   
4             0          0              0   
..          ...        ...            ...   
67            0          0              0   
68            0          0              0   
69            0          0              0   
70            0          0              0   
71            0          0              0   

    DelawareNorthSportserviceandChick-fil-a  FredMyeir  FreshDirectTerrace  \
0                                         0          0                   0   
1                                         0          0                   0   
2                                         0          0                   0   
3                                         0          0                   0   
4                                         0          0                   0   
..                                      ...        ...                 ...   
67                                        0          0                   0   
68                                        0          0                   0   
69                                        0          0                   0   
70                                        0          0                   0   
71                                        0          0                   0   

    FrysFoodStore  Getir  Giant  GiantEagle  H-E-B  JuanValdezCoffee  \
0               0      0      0           0      1                 0   
1               0      0      0           0      0                 0   
2               0      1      0           0      0                 0   
3               0      0      1           0      0                 0   
4               0      0      0           1      0                 0   
..            ...    ...    ...         ...    ...               ...   
67              0      0      0           0      0                 0   
68              0      0      0           0      0                 0   
69              0      0      0           0      0                 0   
70              0      0      0           0      1                 0   
71              0      0      0           0      0                 0   

    KraftHeinz  Kroger  Kroker  LLC  LevyRestaurants  MYNDDRINKS  McDonald's  \
0            0       0       0    0                0           0           0   
1            0       0       0    0                0           0           0   
2            0       0       0    0                0           0           0   
3            0       0       0    0                0           0           0   
4            0       0       0    0                0           0           0   
..         ...     ...     ...  ...              ...         ...         ...   
67           0       0       0    0                0           0           0   
68           0       0       0    0                0           0           0   
69           0       0       0    0                0           0           0

In [49]:
mlb_grocery_partner = dummies.loc[:, 'AramarkSports+Entertainment':'andJuiceItUp!']
mlb_grocery_partner = pd.concat([mlb_grocery_partner, mlb2.team_name], axis=1)
mlb_grocery_partner = pd.concat([mlb_grocery_partner, mlb2.grocery_partner], axis=1)

In [50]:
value_counts2=[]
for i in range(len(mlb_grocery_partner.columns)-2):
    value_counts2.append(mlb_grocery_partner.iloc[:,i].value_counts())
    print(mlb_grocery_partner.iloc[:,i].value_counts())

AramarkSports+Entertainment
0    71
1     1
Name: count, dtype: int64
BonAppétitManagementCompany
0    71
1     1
Name: count, dtype: int64
CHSInc
0    71
1     1
Name: count, dtype: int64
Chick-Fli-A
0    71
1     1
Name: count, dtype: int64
Coca-Cola
0    70
1     2
Name: count, dtype: int64
DelawareNorth
0    71
1     1
Name: count, dtype: int64
DelawareNorthSportserviceandChick-fil-a
0    71
1     1
Name: count, dtype: int64
FredMyeir
0    71
1     1
Name: count, dtype: int64
FreshDirectTerrace
0    71
1     1
Name: count, dtype: int64
FrysFoodStore
0    71
1     1
Name: count, dtype: int64
Getir
0    71
1     1
Name: count, dtype: int64
Giant
0    71
1     1
Name: count, dtype: int64
GiantEagle
0    71
1     1
Name: count, dtype: int64
H-E-B
0    70
1     2
Name: count, dtype: int64
JuanValdezCoffee
0    71
1     1
Name: count, dtype: int64
KraftHeinz
0    71
1     1
Name: count, dtype: int64
Kroger
0    71
1     1
Name: count, dtype: int64
Kroker
0    71
1     1
Name: count, dtyp

In [51]:
charts2 = []
for i in range(len(value_counts2)):
    value_counts = value_counts2[i].reset_index()
    value_counts.columns = ['category', 'count']
    charts2.append(alt.Chart(value_counts,title='Histogram of Food Sponsors').mark_bar().encode(
        x=alt.X('category:N', title='Does This Grocery Partner sponsor this stadium?'),
        y=alt.Y('count:Q', title='Count of Records'),
    ))

In [52]:
print(value_counts2[0])
charts2[0]

AramarkSports+Entertainment
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [53]:
print(value_counts2[1])
(charts2[1])

BonAppétitManagementCompany
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [54]:
print(value_counts2[2])
charts2[2]

CHSInc
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [55]:
print(value_counts2[3])
(charts2[3])

Chick-Fli-A
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [56]:
print(value_counts2[4])
charts2[4]

Coca-Cola
0    70
1     2
Name: count, dtype: int64


alt.Chart(...)

In [57]:
print(value_counts2[5])
(charts2[5])

DelawareNorth
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [58]:
print(value_counts2[6])
charts2[6]

DelawareNorthSportserviceandChick-fil-a
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [59]:
print(value_counts2[7])
(charts2[7])

FredMyeir
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [60]:
print(value_counts2[8])
charts2[8]

FreshDirectTerrace
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [61]:
print(value_counts2[9])
(charts2[9])

FrysFoodStore
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [62]:
print(value_counts2[10])
charts2[10]

Getir
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [63]:
print(value_counts2[11])
(charts2[11])

Giant
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [64]:
print(value_counts2[12])
charts2[12]

GiantEagle
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [65]:
print(value_counts2[13])
(charts2[13])

H-E-B
0    70
1     2
Name: count, dtype: int64


alt.Chart(...)

In [66]:
print(value_counts2[14])
charts2[14]

JuanValdezCoffee
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [67]:
print(value_counts2[15])
(charts2[15])

KraftHeinz
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [68]:
print(value_counts2[16])
charts2[16]

Kroger
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [69]:
print(value_counts2[17])
(charts2[17])

Kroker
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [70]:
print(value_counts2[18])
charts2[18]

LLC
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [71]:
print(value_counts2[19])
(charts2[19])

LevyRestaurants
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [72]:
print(value_counts2[20])
charts2[20]

MYNDDRINKS
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [73]:
print(value_counts2[21])
(charts2[21])

McDonald's
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [74]:
print(value_counts2[22])
charts2[22]

Meijer
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [75]:
print(value_counts2[23])
(charts2[23])

MountainMike'sPizza
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [76]:
print(value_counts2[24])
charts2[24]

Old-Timers'DaySponsorship
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [77]:
print(value_counts2[25])
(charts2[25])

OnTheBorder
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [78]:
print(value_counts2[26])
charts2[26]

PriceChopper
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [79]:
print(value_counts2[27])
(charts2[27])

Raley's
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [80]:
print(value_counts2[28])
charts2[28]

SCJohnson
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [81]:
print(value_counts2[29])
(charts2[29])

Safeway
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [82]:
print(value_counts2[30])
charts2[30]

SchnuckMarket
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [83]:
print(value_counts2[31])
(charts2[31])

Sedano'sSupermarketsChain30+locations
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [84]:
print(value_counts2[32])
charts2[32]

Smart&Final
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [85]:
print(value_counts2[33])
(charts2[33])

Sobeys
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [86]:
print(value_counts2[34])
charts2[34]

Spec's
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [87]:
print(value_counts2[35])
(charts2[35])

StopandShopSupermarketCompany
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [88]:
print(value_counts2[36])
charts2[36]

Sugardale
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [89]:
print(value_counts2[37])
(charts2[37])

Tex-Mex
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [90]:
print(value_counts2[38])
charts2[38]

ViennaBeef
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [91]:
print(value_counts2[39])
(charts2[39])

ZENB
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [92]:
print(value_counts2[40])
charts2[40]

aHouston-basedliquorretailer
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [93]:
print(value_counts2[41])
(charts2[41])

andCommunityOutreach
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [94]:
print(value_counts2[42])
charts2[42]

andFeedingTampaBay
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

In [95]:
print(value_counts2[43])
(charts2[43])

andJuiceItUp!
0    71
1     1
Name: count, dtype: int64


alt.Chart(...)

The most common food distributor across venueons are Delaware North and Sysco, whereas the most common grocery partners across venueons are the H-E-B and Giant Eagle

What are the most popular food categories available for sponsorship? Are there gaps (e.g., niche categories like brisket or smoked chicken)

In [96]:
mlb2 = drop_duplicate_columns(mlb2)

In [97]:
mlb2['available_food_categories'] = mlb2['available_food_categories'].str.replace(' ','')
# One-hot encoding with comma as a separator
dummies = mlb2['available_food_categories'].str.get_dummies(sep=',')
# Concatenate the encoded columns with the original DataFrame
mlb2 = pd.concat([mlb2, dummies], axis=1)

In [98]:
mlb2 = drop_duplicate_columns(mlb2)

In [99]:
mlb_foods = dummies.iloc[:,0:len(dummies)-1]
mlb_foods = pd.concat([mlb_foods, mlb2.team_name], axis=1)
mlb_foods = pd.concat([mlb_foods, mlb2.available_food_categories], axis=1)

In [100]:
for i in range(36):
    print(i,mlb_foods.iloc[:,i].value_counts())

0 All-starandothernewburgers
0    71
1     1
Name: count, dtype: int64
1 BBQ
0    69
1     3
Name: count, dtype: int64
2 BBQopenHotdogssausageandbaconalltaken
0    71
1     1
Name: count, dtype: int64
3 Ballparkfood
0    68
1     4
Name: count, dtype: int64
4 Beef
0    66
1     6
Name: count, dtype: int64
5 BerbecueSauce
0    66
1     6
Name: count, dtype: int64
6 Brats
0    69
1     3
Name: count, dtype: int64
7 Brisket
0    66
1     6
Name: count, dtype: int64
8 BuccoCrunchDog
0    71
1     1
Name: count, dtype: int64
9 ChickenWings
0    60
1    12
Name: count, dtype: int64
10 FriedCheeseCurds
0    71
1     1
Name: count, dtype: int64
11 Fries
0    71
1     1
Name: count, dtype: int64
12 Hamburgers
0    62
1    10
Name: count, dtype: int64
13 HighlightedSectionsareteamsthatdonthaveaBBQSponsor
0    71
1     1
Name: count, dtype: int64
14 HotDogs
0    51
1    21
Name: count, dtype: int64
15 Ice-Cream
0    58
1    14
Name: count, dtype: int64
16 Mrs.T'sToppedPierogies
0    71
1     1
Na

The most popular food categories available for sponsorships are Hot Dogs, Pizza, Ice Cream, and Salty Snacks.  There are several gaps, including (Traditional) Ballpark food, Popcorn, Soft Drinks, Veggies and Vegan, and Tacos.

Are there opportunities in categories where sponsorship is low but concession volume is high

In [101]:
# Remove spaces from the strings
mlb2['concession_tray_annual_volume'] = mlb2['concession_tray_annual_volume'].str.replace(' ','') + "_concession_vol"
# One-hot encoding with comma as a separator
dummies = mlb2['concession_tray_annual_volume'].str.get_dummies(sep=',')
# Concatenate the encoded columns with the original DataFrame
mlb2 = pd.concat([mlb2, dummies], axis=1)
mlb2 = drop_duplicate_columns(mlb2)

In [102]:
mlb2

team_name           timestamp      contact_name  \
0          Houston Astros  7/26/2023 17:33:15    Jimmy Comerota   
1      Kansas City Royals    6/5/2025 5:50:32       David Adams   
2           New York Mets    6/5/2025 6:08:11     Meghan Baker    
3   Philadelphia Phillies    6/5/2025 6:14:42     Scott Nickie    
4      Pittsburgh Pirates    6/5/2025 8:18:27     Chris Stevens   
..                    ...                 ...               ...   
67                 unsure              unsure            unsure   
68                 unsure              unsure            unsure   
69                 unsure              unsure            unsure   
70         Houston Astros    6/5/2025 5:45:15    Jimmy Comerota   
71       San Diego Padres  2/18/2025 14:12:36  Sergio Del Prado   

      phone_number grocery_partner concession_corp  \
0   (713) 447-0782           H-E-B         Aramark   
1     760-424-6067    PriceChopper         Aramark   
2     718-559-3047           Getir         Aramark   
3     215-218-5286           Giant         Aramark   
4     412-260-5559      GiantEagle         Aramark   
..             ...             ...             ...   
67          unsure          unsure          unsure   
68          unsure          unsure          unsure   
69          unsure          unsure          unsure   
70  (713)-447-0782           H-E-B         Aramark   
71  (619) 208-7038          unsure  Delaware North   

                                         manager_name  \
0   He felt uncomfortable during the phone call co...   
1                                       Matt Quatraro   
2                                      Carlos Mendoza   
3                                         Rob Thomson   
4   John Bray, Manager, Corporate & Premium Partne...   
..                                                ...   
67                                             unsure   
68                                             unsure   
69                                             unsure   
70                                             unsure   
71                                             unsure   

                      food_distributor    convenience_partner  \
0                                Sysco                 unsure   
1                      KauffmanStadium               QuikTrip   
2                            CitiField                 unsure   
3                  GiantFoodStores,LLC                  Giant   
4   GreaterPittsburghCommunityFoodBank  Multiple-No Exclusive   
..                                 ...                    ...   
67                              unsure                 unsure   
68                              unsure                 unsure   
69                              unsure                 unsure   
70                               Sysco                 unsure   
71                               Sysco                 unsure   

   has_cuphold_sponsors cuphold_sponsors  \
0                    No           unsure   
1                    No           unsure   
2                    No           unsure   
3                   Yes          Citizen   
4                    No           unsure   
..                  ...              ...   
67               unsure           unsure   
68               unsure           unsure   
69               unsure           unsure   
70                   No           unsure   
71                   No           unsure   

                                concession_area_items has_wifi  \
0   Souvenir cups, Concession trays, printed paper...      Yes   
1   Souvenir cups, Concession trays, printed paper...      Yes   
2                                       Souvenir cups      Yes   
3        cup holders, Souvenir cups, Concession trays      Yes   
4                     Souvenir cups, Concession trays      Yes   
..                                                ...      ...   
67                                             unsure   unsure   
68                                     

In [103]:
mlb_volumes = mlb2.loc[:,'Ateveryareawheretheyhavefood_concession_vol':'unsure_concession_vol']
mlb_volumes = pd.concat([mlb_volumes, mlb2.team_name], axis=1)
mlb_volumes = pd.concat([mlb_volumes, mlb2.concession_tray_annual_volume], axis=1)

In [104]:
mlb_volumes

Ateveryareawheretheyhavefood_concession_vol  \
0                                            0   
1                                            0   
2                                            0   
3                                            1   
4                                            1   
..                                         ...   
67                                           0   
68                                           0   
69                                           0   
70                                           0   
71                                           0   

   Ateveryfoodcourtarea_concession_vol  \
0                                    0   
1                                    0   
2                                    0   
3                                    0   
4                                    0   
..                                 ...   
67                                   0   
68                                   0   
69                                   0   
70                                   0   
71                                   0   

   Ateverylocationwherethereisfood_concession_vol unsure_concession_vol  \
0                                               0                     1   
1                                               1                     0   
2                                               0                     1   
3                                               0                     0   
4                                               0                     0   
..                                            ...                   ...   
67                                              0                     1   
68                                              0                     1   
69                                              0                     1   
70                                              0                     1   
71                                              0                     1   

                team_name                   concession_tray_annual_volume  
0          Houston Astros                           unsure_concession_vol  
1      Kansas City Royals  Ateverylocationwherethereisfood_concession_vol  
2           New York Mets                           unsure_concession_vol  
3   Philadelphia Phillies     Ateveryareawheretheyhavefood_concession_vol  
4      Pittsburgh Pirates     Ateveryareawheretheyhavefood_concession_vol  
..                    ...                                             ...  
67                 unsure                           unsure_concession_vol  
68                 unsure                           unsure_concession_vol  
69                 unsure                           unsure_concession_vol  
70         Houston Astros                           unsure_concession_vol  
71       San Diego Padres                           unsure_concession_vol  

[72 rows x 6 columns]

In [105]:
mlb_volumes = drop_duplicate_columns(mlb_volumes)

In [106]:
mlb_volumes

Ateveryareawheretheyhavefood_concession_vol  \
0                                            0   
1                                            0   
2                                            0   
3                                            1   
4                                            1   
..                                         ...   
67                                           0   
68                                           0   
69                                           0   
70                                           0   
71                                           0   

   Ateveryfoodcourtarea_concession_vol  \
0                                    0   
1                                    0   
2                                    0   
3                                    0   
4                                    0   
..                                 ...   
67                                   0   
68                                   0   
69                                   0   
70                                   0   
71                                   0   

   Ateverylocationwherethereisfood_concession_vol unsure_concession_vol  \
0                                               0                     1   
1                                               1                     0   
2                                               0                     1   
3                                               0                     0   
4                                               0                     0   
..                                            ...                   ...   
67                                              0                     1   
68                                              0                     1   
69                                              0                     1   
70                                              0                     1   
71                                              0                     1   

                team_name                   concession_tray_annual_volume  
0          Houston Astros                           unsure_concession_vol  
1      Kansas City Royals  Ateverylocationwherethereisfood_concession_vol  
2           New York Mets                           unsure_concession_vol  
3   Philadelphia Phillies     Ateveryareawheretheyhavefood_concession_vol  
4      Pittsburgh Pirates     Ateveryareawheretheyhavefood_concession_vol  
..                    ...                                             ...  
67                 unsure                           unsure_concession_vol  
68                 unsure                           unsure_concession_vol  
69                 unsure                           unsure_concession_vol  
70         Houston Astros                           unsure_concession_vol  
71       San Diego Padres                           unsure_concession_vol  

[72 rows x 6 columns]

In [107]:
pd.unique(mlb_volumes.columns)

array(['Ateveryareawheretheyhavefood_concession_vol',
       'Ateveryfoodcourtarea_concession_vol',
       'Ateverylocationwherethereisfood_concession_vol',
       'unsure_concession_vol', 'team_name',
       'concession_tray_annual_volume'], dtype=object)

In [108]:
mlb2= mlb2.T.drop_duplicates().T

In [109]:
mlb2 = mlb2.drop(columns={"Coca-Cola","AramarkSports+Entertainment",'unsure'})

In [110]:
alt.Chart(mlb2,title="Bar Chart of concession volumes").mark_bar().encode(
    x=alt.X("concession_tray_annual_volume:N"),
    y=alt.Y("count():Q").scale(zero=False),
)

alt.Chart(...)

Yes.  There are several categories where sponsorship is low but concession volume is high in multiple venueons.  These categories include non-alcholic beverages (e.g. Coca Cola, Soft Drinks, etc.), Salty Snacks (e.g Popcorn, peanuts, chips, pretzels, and candy), specialty beverages (including coffee and iced drinks), Frozen Treats (e.g Ice Cream), and Healthy Alternatives (Including Veggie/Vegan).  These Categories are attractive because of the following reasons:
- Consumer captive audience: MLB games have long dwell times (~3 hours), increasing purchase frequency.
- Emotional context: Positive associations during games can drive brand affinity.
- Lack of competition: In under-sponsored categories, a brand can quickly become synonymous with the product in that environment.
- Scalable: Multi-team or league-wide deals can be negotiated if successful in one market.

This Answer is Curteousy of ChatGPT.

4. What is the average stadium seat count compared to souvenir cup distribution? Can increased souvenir cups lead to more ad impressions

In [111]:
seats = 0
answers = 0
for i in range(len(mlb2)):
    if mlb2.at[i,'cupholders_per_seat'] != 'unsure':
        number = pd.to_numeric("".join(filter(str.isdigit, mlb2.at[i,'cupholders_per_seat'])))
        mlb2.at[i,'cupholders_per_seat_numeric'] = number
        seats += number
        answers += 1
    else:
        mlb2.at[i,'cupholders_per_seat_numeric'] = 0

In [112]:
# Remove spaces from the strings
mlb2['pouring_partners'] = mlb2['pouring_partners'].str.replace(' ','')
# One-hot encoding with comma as a separator
dummies = mlb2['pouring_partners'].str.get_dummies(sep=',')
# Concatenate the encoded columns with the original DataFrame
mlb2 = pd.concat([mlb2, dummies], axis=1)

In [113]:
mlb2

team_name           timestamp      contact_name  \
0          Houston Astros  7/26/2023 17:33:15    Jimmy Comerota   
1      Kansas City Royals    6/5/2025 5:50:32       David Adams   
2           New York Mets    6/5/2025 6:08:11     Meghan Baker    
3   Philadelphia Phillies    6/5/2025 6:14:42     Scott Nickie    
4      Pittsburgh Pirates    6/5/2025 8:18:27     Chris Stevens   
..                    ...                 ...               ...   
67                 unsure              unsure            unsure   
68                 unsure              unsure            unsure   
69                 unsure              unsure            unsure   
70         Houston Astros    6/5/2025 5:45:15    Jimmy Comerota   
71       San Diego Padres  2/18/2025 14:12:36  Sergio Del Prado   

      phone_number grocery_partner concession_corp  \
0   (713) 447-0782           H-E-B         Aramark   
1     760-424-6067    PriceChopper         Aramark   
2     718-559-3047           Getir         Aramark   
3     215-218-5286           Giant         Aramark   
4     412-260-5559      GiantEagle         Aramark   
..             ...             ...             ...   
67          unsure          unsure          unsure   
68          unsure          unsure          unsure   
69          unsure          unsure          unsure   
70  (713)-447-0782           H-E-B         Aramark   
71  (619) 208-7038          unsure  Delaware North   

                                         manager_name  \
0   He felt uncomfortable during the phone call co...   
1                                       Matt Quatraro   
2                                      Carlos Mendoza   
3                                         Rob Thomson   
4   John Bray, Manager, Corporate & Premium Partne...   
..                                                ...   
67                                             unsure   
68                                             unsure   
69                                             unsure   
70                                             unsure   
71                                             unsure   

                      food_distributor    convenience_partner  \
0                                Sysco                 unsure   
1                      KauffmanStadium               QuikTrip   
2                            CitiField                 unsure   
3                  GiantFoodStores,LLC                  Giant   
4   GreaterPittsburghCommunityFoodBank  Multiple-No Exclusive   
..                                 ...                    ...   
67                              unsure                 unsure   
68                              unsure                 unsure   
69                              unsure                 unsure   
70                               Sysco                 unsure   
71                               Sysco                 unsure   

   has_cuphold_sponsors cuphold_sponsors  \
0                    No           unsure   
1                    No           unsure   
2                    No           unsure   
3                   Yes          Citizen   
4                    No           unsure   
..                  ...              ...   
67               unsure           unsure   
68               unsure           unsure   
69               unsure           unsure   
70                   No           unsure   
71                   No           unsure   

                                concession_area_items has_wifi  \
0   Souvenir cups, Concession trays, printed paper...      Yes   
1   Souvenir cups, Concession trays, printed paper...      Yes   
2                                       Souvenir cups      Yes   
3        cup holders, Souvenir cups, Concession trays      Yes   
4                     Souvenir cups, Concession trays      Yes   
..                                                ...      ...   
67                                             unsure   unsure   
68                                     

In [114]:
mlb_drinks= dummies.iloc[:,0:len(dummies-1)]
mlb_drinks = pd.concat([mlb_drinks, mlb2.cupholders_per_seat_numeric], axis=1)
mlb_drinks = pd.concat([mlb_drinks, mlb2.team_name], axis=1)

In [115]:
mlb_drinks

Anheuser-Busch  Coca-Cola  Dr.Pepper  MillerCoors  Pepsi  unsure  \
0                0          0          0            0      0       1   
1                0          1          0            1      1       0   
2                1          0          1            0      1       0   
3                1          1          0            1      0       0   
4                1          1          0            1      0       0   
..             ...        ...        ...          ...    ...     ...   
67               0          0          0            0      0       1   
68               0          0          0            0      0       1   
69               0          0          0            0      0       1   
70               0          0          0            0      1       0   
71               0          1          0            0      0       0   

    cupholders_per_seat_numeric              team_name  
0                           0.0         Houston Astros  
1                           0.0     Kansas City Royals  
2                           0.0          New York Mets  
3                       42901.0  Philadelphia Phillies  
4                           0.0     Pittsburgh Pirates  
..                          ...                    ...  
67                          0.0                 unsure  
68                          0.0                 unsure  
69                          0.0                 unsure  
70                      41000.0         Houston Astros  
71                          0.0       San Diego Padres  

[72 rows x 8 columns]

In [116]:
mlb_drinks=mlb_drinks.rename(columns={'Dr.Pepper':'DrPepper'})

In [117]:
alt.Chart(mlb_drinks,title="Bar Chart of Coca-Cola in seats per venue").mark_bar().encode(
    x=alt.X("team_name:N",title="Team Name").scale(zero=False),
    y=alt.Y("cupholders_per_seat_numeric:Q").scale(zero=False),
    color=alt.Color(
        "Coca-Cola:N",title="Coca-Cola"
    )
)

alt.Chart(...)

In [118]:
alt.Chart(mlb_drinks,title="Bar Chart of Pepsi in seats per venue").mark_bar().encode(
    x=alt.X("team_name:N",title="Team Name").scale(zero=False),
    y=alt.Y("cupholders_per_seat_numeric:Q").scale(zero=False),
    color=alt.Color(
        "Pepsi:N",title="Pepsi"
    )
)

alt.Chart(...)

In [119]:
alt.Chart(mlb_drinks,title="Bar Chart of Miller Coors Drinks in seats per venue").mark_bar().encode(
    x=alt.X("team_name:N",title="Team Name").scale(zero=False),
    y=alt.Y("cupholders_per_seat_numeric:Q").scale(zero=False),
    color=alt.Color(
        "MillerCoors:N",title="Miller Coors"
    )
)

alt.Chart(...)

In [120]:
alt.Chart(mlb_drinks,title="Bar Chart of Anheuser-Busch in seats per venue").mark_bar().encode(
    x=alt.X("team_name:N",title="Team Name").scale(zero=False),
    y=alt.Y("cupholders_per_seat_numeric:Q").scale(zero=False),
    color=alt.Color(
        "Anheuser-Busch:N",title="Anheuser-Busch"
    )
)

alt.Chart(...)

In [121]:
alt.Chart(mlb_drinks,title="Bar Chart of Anheuser-Busch in seats per venue").mark_bar().encode(
    x=alt.X("team_name:N",title="Team Name").scale(zero=False),
    y=alt.Y("cupholders_per_seat_numeric:Q").scale(zero=False),
    color=alt.Color(
        "DrPepper:N",title="Dr. Pepper"
    )
)

alt.Chart(...)

In [122]:
print(f"{(seats/answers)/5}")

8217.364705882352


The Average Stadium Seat count is around 41,000 seats per stadium.  When compared to the souvenir cup distribution, the average stadium seat count is around 8,217 seats for each unique souvenir cup in every stadium.

As for ad impressions, increased use of souvenir cups can lead to more ad impressions in MLB baseball stadiums — and it can be a highly effective, underutilized advertising channel.

Here’s why:
1. High Visibility & Retention
- - Souvenir cups are kept longer than standard paper or plastic cups — often taken home as collectibles.

- - This extends the life of the brand impression beyond the game itself, increasing total impressions per unit.

- - They are prominently visible during games, often placed on cupholders, held in fan photos, or caught on camera.

2. Guaranteed Engagement
  - Every cup is physically touched and used by a fan, guaranteeing the brand message is seen.

  - Compared to digital ads or signs, this is unskippable and personal.

3. Mobile Billboards Inside the Venue
- - Fans walk around with cups during the game — in lines, concourses, and seats.
  - This movement makes each cup a mini mobile ad unit, increasing intra-venue exposure.

4. Enhanced Brand Association
- - A branded cup connected to a positive fan experience (like a home run or a great game) increases emotional value and memory.
  - Souvenir cups often have team branding, and advertisers can benefit from a halo effect by being part of that packaging.

5. Monetization & Sponsorship Opportunities
- - Teams can sell cup sponsorships to beverage brands, QSRs, or other companies.
  - Co-branding opportunities (e.g. "Coca-Cola presents the Atlanta Braves 2025 Cup Collection") can drive sponsorship revenue.

This Answer is Curteousy of ChatGPT.